src/Microsoft.ApiDesignGuidelines.Analyzers/UnitTests/UsePropertiesWhereAppropriateTests.cs

```diff
@@ -1,23 +1,415 @@
+using System;
+using System.Collections;
+
+public class GenericType<T>
+{
+}
+
+public class Base
+{
+    public virtual int GetSomething()
+    {
+        return 0;
+    }
+
+    public virtual int GetOverloadedMethod()
+    {
+        return 1;
+    }
+
+    public virtual int GetOverloadedMethod(int i)
+    {
+        return i;
+    }
+}
+
+public class Class1 : Base
+{
+    private string fileName = """";
+
+    // 1) Returns void
+    public void GetWronglyNamedMethod()
+    {
+    }
+
+    // 2) Not a method
+    public string LogFile
+    {
+        get { return fileName; }
+    }
+
+    // 3) Returns an array type
+    public int[] GetValues()
+    {
+        return null;
+    }
+
+    // 4) Has parameters
+    public int[] GetMethodWithParameters(int p)
+    {
+        return new int[] { p };
+    }
+
+    // 5a) Name doesn't start with a 'Get'
+    public int SomeMethod()
+    {
+        return 0;
+    }
+
+    // 5b) First compound word is not 'Get'
+    public int GetterMethod()
+    {
+        return 0;
+    }
+
+    // 6) Generic method
+    public object GetGenericMethod<T>()
+    {
+        return new GenericType<T>();
+    }
+
+    // 7) Override
+    public override int GetSomething()
+    {
+        return 1;
+    }
+
+    // 8) Method with overloads
+    public override int GetOverloadedMethod()
+    {
+        return 1;
+    }
+
+    public override int GetOverloadedMethod(int i)
+    {
+        return i;
+    }
+
+    // 9) Methods with special name
+    public override int GetHashCode()
+    {
+        return 0;
+    }
+
+    public IEnumerator GetEnumerator()
+    {
+        return null;
+    }
+
+    // 10) Method with invocation expressions
+    public int GetSomethingWithInvocation()
+    {
+        Console.WriteLine(this);
+        return 0;
+    }
+
+    // 11) Method named 'Get'
+    public string Get()
+    {
+        return fileName;
+    }
+
+    // 12) Private method
+    private string GetSomethingPrivate()
+    {
+        return fileName;
+    }
+
+    // 13) Internal method
+    internal string GetSomethingInternal()
+    {
+        return fileName;
+    }
+}
+");
+        }
+
+        [Fact]
+        public void CSharp_CA1024DiagnosticCases()
+        {
+            VerifyCSharp(@"
+public class Class
+{
+    private string fileName = ""data.txt"";
+
+    public string GetFileName()
+    {
+        return fileName;
+    }
+
+    public string Get_FileName2()
+    {
+        return fileName;
+    }
+
+    public string Get123()
+    {
+        return fileName;
+    }
+
+    protected string GetFileNameProtected()
+    {
+        return fileName;
+    }
+}
+",
+            GetCA1024CSharpResultAt(6, 19, "GetFileName"),
+            GetCA1024CSharpResultAt(11, 19, "Get_FileName2"),
+            GetCA1024CSharpResultAt(16, 19, "Get123"),
+            GetCA1024CSharpResultAt(21, 22, "GetFileNameProtected"));
+        }
+
+        [Fact]
+        public void CSharp_CA1024DiagnosticCasesWithScope()
+        {
```

Comment:
Remove this test. We no longer use the old analysis API that took a document and a span.